### PR TITLE
VscPreview extension

### PR DESCRIPTION
**Pull request by 928PJY**
structure of the extension
@hellosnow @qinezh @Huachao @superyyrrzz @chenkennt

### CODE REVIEW

File: src/VscPreviewExtension/package.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
+  "name": "DfmPreview",
+  "displayName": "DfmPreview",
+  "description": "Preview the DocFX Flavored Markdown in vscode",
```
Comment (qinezh) [inline src/VscPreviewExtension/package.json]: >vscode [](start = 57, length = 6)

Visual Studio Code

File: src/VscPreviewExtension/DfmParse/.gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+*.dll
+*.exe
+*.config
```
Comment (qinezh) [inline src/VscPreviewExtension/DfmParse/.gitignore]: merge with .gitignore in VscPreviewExtension

File: src/VscPreviewExtension/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+DocFX Flavored Markdown Preview
+==========================================
+
+[![Current Version](http://vsmarketplacebadge.apphb.com/version/928pjy.DfmPreview.svg)](http://marketplace.visualstudio.com/items?itemName=928pjy.DfmPreview)
+[![Install Count](http://vsmarketplacebadge.apphb.com/installs/928pjy.DfmPreview.svg)](https://marketplace.visualstudio.com/items?itemName=928pjy.DfmPreview)
+[![Open Issues](http://vsmarketplacebadge.apphb.com/rating/928pjy.DfmPreview.svg) ](https://marketplace.visualstudio.com/items?itemName=928pjy.DfmPreview)
+> **Notice:** Homepage for the Dfm_Preview extension at [Dfm_Preview](https://github.com/928PJY/Dfm_preview)
```
Comment (qinezh) [inline src/VscPreviewExtension/README.md]: >Dfm_Preview [](start = 31, length = 11)

DfmPreview?

**Comment by qinezh** [comment]
:shipit:

File: src/VscPreviewExtension/DfmParse/.gitignore (hunk 2 above)
Comment (928PJY) [inline src/VscPreviewExtension/DfmParse/.gitignore]: But I want to upload the empty 'DfmParse' folder for that some .exe and .dll  files will be copied to this folder from ‘src/Microsoft.DocAsCode.Dfm.VscPreview/' . Is there anothor way to achieve this. 

---
In reply to: [74867922](https://github.com/dotnet/docfx/pull/609#discussion_r74867922) [](ancestors = 74867922)

Comment (qinezh) [inline src/VscPreviewExtension/DfmParse/.gitignore]: you can create a empty folder named DfmParse before you copy files into this folder

File: src/VscPreviewExtension/.gitignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
+node_modules/
+out/
+DfmParse/
```
Comment (hellosnow) [inline src/VscPreviewExtension/.gitignore]: >DfmParse [](start = 0, length = 8)

there's no DfmParse folder currently, remove and add it back later?

File: src/VscPreviewExtension/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+### 0.0.9
+* Bug fix: can't preview a single file(not in a open folder)
+* Bug fix: throw an error if it is not a markdown file
+
+### 0.0.8
+* Bug fix: open other preview window
+* Bug fix: some css and javasript files to show the html
+
+
+### 0.0.7
```
Comment (hellosnow) [inline src/VscPreviewExtension/README.md]: >### [](start = 0, length = 3)

remove extra line before

File: src/VscPreviewExtension/package.json [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  "main": "./out/src/extension",
+  "contributes": {
+    "commands": [
+      {
+        "command": "DFM.showPreview",
+        "title": "Toggle Dfm Preview"
+      },
+      {
+        "command": "DFM.showPreviewToside",
+        "title": "Open Dfm preview to the side"
```
Comment (superyyrrzz) [inline src/VscPreviewExtension/package.json]: >preview [](start = 27, length = 7)

Preview

File: src/VscPreviewExtension/README.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+### 0.0.8
+* Bug fix: open other preview window
+* Bug fix: some css and javasript files to show the html
+
+### 0.0.7
+* Initial Release!
+
+## Feedback
+Please provide feedback through the [Github Issue](https://github.com/928PJY/Dfm_preview/issues) system , or fork the repository and submit PR
```
Comment (superyyrrzz) [inline src/VscPreviewExtension/README.md]: >  [](start = 103, length = 1)

remove

**Comment by superyyrrzz** [comment]
:shipit:

File: src/VscPreviewExtension/README.md (hunk 7 above)
Comment (qinezh) [inline src/VscPreviewExtension/README.md]: >https://github.com/928PJY/Dfm_preview/issues [](start = 51, length = 44)

update these url? as this extension would be in docfx repo later on

File: src/VscPreviewExtension/README.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
+DocFX Flavored Markdown Preview
+==========================================
+
+[![Current Version](http://vsmarketplacebadge.apphb.com/version/928pjy.DfmPreview.svg)](http://marketplace.visualstudio.com/items?itemName=928pjy.DfmPreview)
+[![Install Count](http://vsmarketplacebadge.apphb.com/installs/928pjy.DfmPreview.svg)](https://marketplace.visualstudio.com/items?itemName=928pjy.DfmPreview)
+[![Open Issues](http://vsmarketplacebadge.apphb.com/rating/928pjy.DfmPreview.svg) ](https://marketplace.visualstudio.com/items?itemName=928pjy.DfmPreview)
+> **Notice:** Homepage for the DfmPreview extension at [DfmPreview](https://github.com/928PJY/Dfm_preview)
```
Comment (qinezh) [inline src/VscPreviewExtension/README.md]: >https://github.com/928PJY/Dfm_preview [](start = 68, length = 37)

also update the url here

**Pull request merged**